metadata-models/src/main/pegasus/com/linkedin/dataaccess/SchemaFieldAccess.pdl

```diff
@@ -0,0 +1,22 @@
+namespace com.linkedin.dataaccess
+
+import com.linkedin.common.TimeStamp
+import com.linkedin.dataset.SchemaFieldPath
+import com.linkedin.schema.SchemaFieldDataType
+
+record SchemaFieldAccess {
+
+  fieldPath: SchemaFieldPath
+
+  type: SchemaFieldDataType
```

Comment:
Good point. `type` here could seem redundant since it is available in the dataset schema as well. But it could happen that the field type in the table is changed by the table owner after an access has been created, which indicates the field with the same name might start to have a different meaning, which might render it inconsistent with the intention of the access.

Of course there could be alternative mechanisms to prevent/resolve such an inconsistency. Nevertheless we are also open to removing this field if there is a good reason.